scripts/pylib/twister/twisterlib/environment.py

```diff
@@ -958,6 +958,22 @@ def __init__(self, options=None, default_options=None) -> None:
             if snippet_root:
                 self.snippet_roots.append(Path(module.project) / snippet_root)
 
+
+        self.soc_roots = [Path(ZEPHYR_BASE), Path(ZEPHYR_BASE) / 'subsys' / 'testsuite']
+        self.dts_roots = [Path(ZEPHYR_BASE)]
+        self.arch_roots = [Path(ZEPHYR_BASE)]
+
+        for module in modules:
+            soc_root = module.meta.get("build", {}).get("settings", {}).get("soc_root")
+            if soc_root:
+                self.soc_roots.append(os.path.join(module.project, soc_root))
+            dts_root = module.meta.get("build", {}).get("settings", {}).get("dts_root")
+            if soc_root:
+                self.dts_roots.append(os.path.join(module.project, dts_root))
+            arch_root = module.meta.get("build", {}).get("settings", {}).get("arch_root")
+            if arch_root:
+                self.arch_roots.append(os.path.join(module.project, arch_root))
+
```

Comment:
**suggestion (if-minor):** What's about `board_root`? AFAIR, I had to hack the Twister testplan.py too. See: https://github.com/zephyrproject-rtos/zephyr/issues/71761#issuecomment-2084427049

Reply:
Common question: Wouldn't it be conceivable / more practical if both West and Twister obtained their information from a central Python code base? West uses similar code in `scripts/west_commands/boards.py`, `class Boards(WestCommand)`. Twister uses `scripts/pylib/twister/twisterlib/environment.py`, `__init__()`.

With a shared code base, the same deficits could then also be solved in the Sphinx Extensions. See https://github.com/zephyrproject-rtos/zephyr/issues/71761#issuecomment-2083500814

Reply:
> **suggestion (if-minor):** What's about `board_root`? AFAIR, I had to hack the Twister testplan.py too. See: [#71761 (comment)](https://github.com/zephyrproject-rtos/zephyr/issues/71761#issuecomment-2084427049)

this is already done here: https://github.com/zephyrproject-rtos/zephyr/blob/cfe5bd922432d9c2090e9cbf149edd7172dc4e62/scripts/pylib/twister/twisterlib/environment.py#L261

> Common question: Wouldn't it be conceivable / more practical if both West and Twister obtained their information from a central Python code base? West uses similar code in `scripts/west_commands/boards.py`, `class Boards(WestCommand)`. Twister uses `scripts/pylib/twister/twisterlib/environment.py`, `__init__()`.
> 
> With a shared code base, the same deficits could then also be solvable in the Sphinx Extensions. See [#71761 (comment)](https://github.com/zephyrproject-rtos/zephyr/issues/71761#issuecomment-2083500814)

we are using common code, in this case the `zephyr_module` module.